### PR TITLE
Move away from recursive go test, as it cannot output a merged coverage profile.

### PR DESCRIPTION
**Pull request by tomwilkie**
- Update said profile to coveralls.io
- Add badge to readme

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,4 +13,9 @@ install:
   - go install -tags netgo std
   - make travis
 
-script: make tests
+script:
+  - make tests
+
+after_success:
+  - go get github.com/mattn/goveralls
+  - $HOME/gopath/bin/goveralls -coverprofile=profile.cov -service=travis-ci
```
Comment (rade) [inline .travis.yml]: newline missing at end of file

File: Makefile [hunk 2]
```diff
@@ -61,7 +61,14 @@ $(DOCKER_DISTRIB):
...
 tests:
-	go test -cover -tags netgo ./...
+	echo "mode: count" > profile.cov
+	for dir in $$(find . -type f -name '*_test.go' | xargs -n1 dirname | sort | uniq); do \
+	    go test -tags netgo -covermode=count -coverprofile=$$dir/profile.tmp $$dir;       \
+	    if [ -f $$dir/profile.tmp ]; then                                                 \
+	        cat $$dir/profile.tmp | tail -n +2 >> profile.cov;                            \
+	        rm $$dir/profile.tmp;                                                         \
+	    fi                                                                                \
+	done
```
Comment (rade) [inline Makefile]: Shouldn't profile.cov be added to .gitignore? Also, what happens to this file? I.e. when I run this manually, how do I look at the results?

**Comment by tomwilkie** [comment]
PTAL

File: Makefile [hunk 3]
```diff
@@ -61,7 +61,15 @@ $(DOCKER_DISTRIB):
...
+	go tool cover -html=profile.cov -o=coverage.html
```
Comment (rade) [inline Makefile]: @dpw mind taking a quick look at the above?

File: Makefile [hunk 4]
```diff
@@ -61,7 +61,15 @@ $(DOCKER_DISTRIB):
 	curl -o $(DOCKER_DISTRIB) $(DOCKER_DISTRIB_URL)
 
 tests:
-	go test -cover -tags netgo ./...
+	echo "mode: count" > profile.cov
+	for dir in $$(find . -type f -name '*_test.go' | xargs -n1 dirname | sort | uniq); do \
```
Comment (dpw) [inline Makefile]: This an be expressed more concisely with something like:

    find . -type f -name '*_test.go' -printf '%h' | sort -u

Comment (tomwilkie) [inline Makefile]: No -printf on mac os x I'm afraid; probably not important for this repo, but thats why its there.  Still want me to change it?

File: Makefile [hunk 5]
```diff
@@ -61,7 +61,15 @@ $(DOCKER_DISTRIB):
...
+	    go test -tags netgo -covermode=count -coverprofile=$$dir/profile.tmp $$dir;       \
+	    if [ -f $$dir/profile.tmp ]; then                                                 \
+	        cat $$dir/profile.tmp | tail -n +2 >> profile.cov;                            \
```
Comment (dpw) [inline Makefile]: No need to invoke cat there:

    tail -n +2 <$$dir/profile.tmp >> profile.cov

Also, does the temp file need to be in the directory?  Better to put it under tmp if it doesn't (e.g. `${TMPDIR:-/tmp}/profile.$$`).

File: Makefile (hunk 4 above)
Comment (dpw) [inline Makefile]: Ok, just the `sort -u` then.

Comment (tomwilkie) [inline Makefile]: Done; thanks!

**Pull request merged**